src/AutoRollerAdapterDeployer.sol

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.15;
+import { AutoRoller, PeripheryLike, OwnedAdapterLike } from "./AutoRoller.sol";
+
+interface OwnableFactoryLike {
+    function rlvFactory() external view returns (address);
+    function deployAdapter(address, bytes memory) external returns (address);
+}
+
+interface AutoRollerFactoryLike {
+    function create(
+        OwnedAdapterLike,
+        address rewardRecipient,
+        uint256 targetDuration
+    ) external returns (AutoRoller);
+}
+
+contract AutoRollerAdapterDeployer {
+    PeripheryLike internal immutable periphery;
```

Comment:
Idea 1: have the `periphery` mutable + setter `setPeriphery(address _periphery)` so we can re-use this contract if the periphery changes?
(we will need to inherit `Trust.sol` so only admin can call this function)

Idea 2: Instead of having an immutable `periphery`, have `divider` and then we can ping the divider to get the current periphery?

Reply:
I went with the later so we don't need to remember to update the deployer whenever we update our periphery!